source/common/secret/secret_manager_impl.cc

```diff
@@ -37,5 +38,37 @@ TlsCertificateConfigProviderSharedPtr SecretManagerImpl::createInlineTlsCertific
   return std::make_shared<TlsCertificateConfigProviderImpl>(tls_certificate);
 }
 
+void SecretManagerImpl::removeDynamicSecretProvider(const std::string& map_key) {
+  ENVOY_LOG(debug, "Unregister secret provider. hash key: {}", map_key);
+
+  if (dynamic_secret_providers_.erase(map_key) == 0) {
```

Comment:
Can this happen? Should this be an assert?

Reply:
This should not happen, added assert for that. Thanks!